src/providers/WalletProvider.tsx

```diff
@@ -33,19 +33,42 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }) => {
     useState<Omit<WalletContextType, "isPending">>(initialState);
   const [isPending, startTransition] = useTransition();
   const popupLock = useRef(false);
-
   const signTransaction = wallet.signTransaction.bind(wallet);
+  const stateRef = useRef(state);
+  useEffect(() => {
```

Comment:
This seems a little odd – @zachfedor could you share some input on your thoughts here?